### ISSUE TITLE
Generalize `mobile-breakpoint` to multiple sizes in `<wa-page>`

### ISSUE THREAD

**Issue by LeaVerou** [shoelace-style/webawesome#244]
https://webawesome-git-next-font-awesome.vercel.app/docs/components/page#responsiveness

Down the line we may want to introduce more breakpoints. What about something like this:

```html
<!-- Now: -->
<wa-page mobile-breakpoint="600">

<!-- After: -->
<wa-page breakpoints="mobile: 600, tablet: 800">

<!-- Or: -->
<wa-page breakpoints="mobile 600, tablet 600">
```

It’s 1-2 characters longer, but gives us a lot more flexibility. We could even accept arbitrary breakpoints and give authors conveniences for handling them.

**Comment by LeaVerou** [comment]
Resolved to the last syntax in today's call.

**Comment by LeaVerou** [comment]
@lindsaym-fa you mentioned Kickstarter backers requested this, could you paste their feedback here if you have an easy way of getting to it? (not the unit stuff, that's discussed in #299)

**Comment by lindsaym-fa** [comment]
From HelpScout 163650 —

> Can't wait to try out wa-page when I have this free weekend. For those of us coming from Bootstrap, I would love to have breakpoints that match up (tiny, small, medium, large, extra-large)

**Comment by LeaVerou** [comment]
Oh they want entirely custom ones, not just to add to `mobile`. Then how would we know when to do the thing the mobile one currently does? 🤔

**Comment by lindsaym-fa** [comment]
I don't think this person is referring to custom breakpoints. I believe they're referring to [the scope of breakpoints that Bootstrap has](https://getbootstrap.com/docs/5.3/layout/breakpoints/). Bootstrap has these options:

- Extra small (default), which they describe as portrait phones
- Small (`sm`), i.e. landscape phones
- Medium (`md`), i.e. tablets
- Large (`lg`), i.e. desktops
- Extra large (`xl`), i.e. large desktops
- Extra extra large (`xxl`), i.e. larger desktops

This person seems to primarily use `sm`, `md`, and `lg` on their site. A selection like `mobile`, `tablet`, and `desktop` (maybe even `laptop` and `tv`?) would likely suffice. (This is a very engaged user, so we could always ask!)

That said, re:"how would we know when to do the thing the mobile one currently does?" I wonder if we should have some way for users to declare which behaviors happen at which breakpoint. E.g., does every user always want the navigation to collapse on `mobile`? With additional breakpoints, what if they'd prefer it collapse on `tablet`?

I'd also love to see some other default behaviors with other slots — i.e., perhaps the aside slot should move below the main slot on the `tablet` breakpoint with its default `flex-direction` reset to `row`. In the owl demo example, I have the aside hidden entirely on mobile, which breaks my heart — I hate hiding content from mobile users. 🥲

**Comment by LeaVerou** [comment]
> I'd also love to see some other default behaviors with other slots — i.e., perhaps the aside slot should move below the main slot on the tablet breakpoint with its default flex-direction reset to row. 

💯 

I wonder if doing these at certain screen sizes is the right way to go though. Presumably, it depends on the navigation size too, right? Container queries could provide more flexibility here…
